config/vector.spec.toml

```diff
@@ -473,19 +473,28 @@ dns_servers = ["0.0.0.0:53"]
   # * type: string
```

Comment:
I strongly suspect I've missed some other documentation that needs updating.  Please just point me in the right direction of anything I missed and I can go and update it.

Reply:
This file, and much of the other documentation, is auto generated. Run `make generate` to build it. For this new source, you will need to create `.meta/sources/socket.toml` and `website/docs/reference/sources/socket.md.erb`. Most of this may be copied from the related tcp and udp files, of course.

Reply:
Okay, per @binarylogic's request I'm going to leave the docs as-is for now.  We'll do a separate commit to update them later.

Reply:
Okay, docs are now updated and ready for review.